### DISCUSSION TITLE
Accessing validation tag parameters of other tags in go-playground/validator

### DISCUSSION THREAD

**Discussion by edwynrrangel** [go-playground/validator#1118] [General]
I'm using the go-playground/validator package to validate a struct in Go. Here is my struct definition:

```
GetLogRequest struct {
	CreatedAtMin      string `query:"created_at_min" validate:"omitempty,datetime=2006-01-02"`
	CreatedAtMax      string `query:"created_at_max" validate:"omitempty,datetime=2006-01-02,date_greater_than=CreatedAtMin"`
}
```

As you can see, the CreatedAtMin and CreatedAtMax fields have a custom date validation that defines the date format with the datetime tag. I'd like to use this same date format in another custom validation function. Here is the skeleton of my custom validation function:

```
func validateDateGreaterThan(fl validator.FieldLevel) bool {

	dateMax, err := time.Parse("2006-01-02", fl.Field().String())
	if err != nil {
		return false
	}

	dateMin, err := time.Parse("2006-01-02", fl.Parent().FieldByName(fl.Param()).String())
	if err != nil {
		return false
	}

	return dateMin.After(dateMax)
}
```

Is there a way to access the parameter of the datetime validation tag of another field from within my custom validation function, such that I can use that format instead of having to hard-code it into the time.Parse function?

Any help or direction would be greatly appreciated. Thank you.

**Comment by deankarn** [comment]
@edwynrrangel have you tried using `fl.Param()`?
